src/value/borrowed.rs

```diff
@@ -70,20 +69,17 @@ impl<'val> BorrowedSymbolToken<'val> {
         }
     }
 
-    /// Decorates the [`BorrowedSymbolToken`] with text.
-    pub fn with_text(mut self, text: &'val str) -> Self {
+    fn with_text_borrowed(mut self, text: &'static str) -> Self {
         self.text = Some(text);
         self
     }
 
-    /// Decorates the [`BorrowedSymbolToken`] with a local ID.
-    pub fn with_local_sid(mut self, local_sid: SymbolId) -> Self {
+    fn with_local_sid_borrrowed(mut self, local_sid: SymbolId) -> Self {
         self.local_sid = Some(local_sid);
         self
     }
 
-    /// Decorates the [`BorrowedSymbolToken`] with an [`BorrowedImportSource`].
-    pub fn with_source(mut self, table: &'val str, sid: SymbolId) -> Self {
+    fn with_source_borrowed(mut self, table: &'static str, sid: SymbolId) -> Self {
```

Comment:
Why do we have `..._borrowed` and the forms below that don't have the suffix?

Reply:
changed this to construct a new `SymbolToken ` for all the `with _...` construction methods.